beps/bep_dht_mutable_torrents.rst

```diff
@@ -0,0 +1,75 @@
+:BEP: ??
+:Title: Decentralized Mutable Torrents
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  Luca Matteis <lmatteis@gmail.com>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 24-Jul-2016
+:Post-History:
+
+
+Abstract
+========
+
+This extension enables torrents to update based on data stored in the BitTorrent DHT [#BEP-44]_, rather than an HTTP server. The publisher has control over when the torrent is updated, and which content it contains, via mutable DHT items. It is similar to BEP 39 [#BEP-39]_, with the difference being that it uses the DHT to notify and be notified about updates of a torrent, hence should provide more decentralized capabilities to both publishers and consumers.
+
+The intention is to allow publishers to serve content that might change over time in a more decentralized fashion. Consumers interested in publishers content only need to know their public key + optional salt. For instance, entities like Archive.org could publish their database dumps using decentralized mutable torrents, and benefit from not having to maintain a central HTTP feed server to notify consumers about updates.
+
+Rationale
+=========
+
+BEP 39 [#BEP-39]_ allows to automatically update torrents based on HTTP URLs. With DHT storing capabilities [#BEP-44]_, we can implement similar functionality in a more decentralized fashion without resorting to central HTTP servers, and using features already present in the BitTorrent network (mainly DHT store).
+
+
+Publishing content
+==================
+
+Publishers should issue a mutable ``put`` request to the DHT when they want to notify consumers about an update of a torrent. The value of the payload ``v`` is the 20 byte infohash of such torrent. Note that there is a 1-to-1 mapping between a mutable item and a torrent, hence a mutable torrent can only ever map to a single torrent.
+
+For request and response details refer to BEP 44 [#BEP-44]_.
+
+Consuming content
+=================
+
+Consumers issue a ``get`` request using the ID of the mutable torrent they are interested in downloading. Periodically polling such ID by issuing ``get`` requests to see whether the ``v`` property of the response has updated. If an update is found, the torrent can be updated using the new infohash.
+
+Both publisher and consumer should periodically ``put`` the mutable items they have active to keep them alive in the DHT.
```

Comment:
I think it may be appropriate to propose an algorithm for determining how frequently to re-announce here. For very popular pieces of content, it could be that having every consumer re-announcing would put too much load on the nodes storing it. I believe @the8472 described an algorithm like this when we originally discussed the put/get protocol for distributed RSS feeds, 5 years ago or so.

basically, if it's possible to estimate how many other peers are re-publishing it, or how frequently it is being republished, one could adjust the probabilities to lower the load.